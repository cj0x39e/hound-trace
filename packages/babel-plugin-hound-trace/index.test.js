const pluginTester = require('babel-plugin-tester');
const houndTrace = require('./index');

pluginTester({
    plugin: houndTrace,
    pluginName: 'hound-trace',
    snapshot: true,
    tests: [
        {
            title: 'function declaration',
            code: `
                function test(a, b, c) {
                    const cj = 'cj';
                }
            `
        },
        {
            title: 'function expression',
            code: `
                const test = function (a, b, c) {
                    const cj = 'cj';
                }
            `
        },
        {
            title: 'arrow function expression',
            code: `
                const test = (a, b, c) => {
                    const cj = 'cj';
                }
            `
        },
        {
            title: 'call expression with function expression',
            code: `
                callback(function () {
                })
            `
        },
        {
            title: 'call expression with arrow function expression',
            code: `
                callback(() => {})
            `
        },
        {
            title: '__NoTraceHook__ function',
            code: `
                function __NoTraceHook__A() {}

                const __NoTraceHook__B = () =>  {}

                const __NoTraceHook__C = function () { }
            `
        },
        {
            title: 'return function',
            code: `
                function A(a, b, c) {
                    if ( a === b) {
                        return 'ok';
                    }

                    switch (c) {
                        case '1':
                            return '1';
                    
                        default:
                            return '2';
                    }

                    return 'error';
                }
            `
        }
    ]
});