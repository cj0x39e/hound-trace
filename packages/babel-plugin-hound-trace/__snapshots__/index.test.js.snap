// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hound-trace __NoTraceHook__ function: __NoTraceHook__ function 1`] = `
"
function __NoTraceHook__A() {}

const __NoTraceHook__B = () =>  {}

const __NoTraceHook__C = function () { }

      ↓ ↓ ↓ ↓ ↓ ↓

function __NoTraceHook__A() {}

const __NoTraceHook__B = () => {};

const __NoTraceHook__C = function () {};
"
`;

exports[`hound-trace arrow function expression: arrow function expression 1`] = `
"
const test = (a, b, c) => {
    const cj = 'cj';
}

      ↓ ↓ ↓ ↓ ↓ ↓

const test = (a, b, c) => {
  let __traceParent__ = window.__traceParent__;
  let __traceOldParent__ = __traceParent__;

  if (window.__trace__) {
    if (!__traceParent__.next) {
      __traceParent__.next = [];
    }

    const current = {
      name: 'test',
      params: [\\"a\\", \\"b\\", \\"c\\"]
    };

    __traceParent__.next.push(current);

    window.__traceParent__ = current;
  }

  const cj = 'cj';
  window.__traceParent__ = __traceOldParent__;
};
"
`;

exports[`hound-trace call expression with arrow function expression: call expression with arrow function expression 1`] = `
"
callback(() => {})

      ↓ ↓ ↓ ↓ ↓ ↓

callback(() => {});
"
`;

exports[`hound-trace call expression with function expression: call expression with function expression 1`] = `
"
callback(function () {
})

      ↓ ↓ ↓ ↓ ↓ ↓

callback(function () {});
"
`;

exports[`hound-trace function declaration: function declaration 1`] = `
"
function test(a, b, c) {
    const cj = 'cj';
}

      ↓ ↓ ↓ ↓ ↓ ↓

function test(a, b, c) {
  let __traceParent__ = window.__traceParent__;
  let __traceOldParent__ = __traceParent__;

  if (window.__trace__) {
    if (!__traceParent__.next) {
      __traceParent__.next = [];
    }

    const current = {
      name: 'test',
      params: [\\"a\\", \\"b\\", \\"c\\"]
    };

    __traceParent__.next.push(current);

    window.__traceParent__ = current;
  }

  const cj = 'cj';
  window.__traceParent__ = __traceOldParent__;
}
"
`;

exports[`hound-trace function expression: function expression 1`] = `
"
const test = function (a, b, c) {
    const cj = 'cj';
}

      ↓ ↓ ↓ ↓ ↓ ↓

const test = function (a, b, c) {
  let __traceParent__ = window.__traceParent__;
  let __traceOldParent__ = __traceParent__;

  if (window.__trace__) {
    if (!__traceParent__.next) {
      __traceParent__.next = [];
    }

    const current = {
      name: 'test',
      params: [\\"a\\", \\"b\\", \\"c\\"]
    };

    __traceParent__.next.push(current);

    window.__traceParent__ = current;
  }

  const cj = 'cj';
  window.__traceParent__ = __traceOldParent__;
};
"
`;

exports[`hound-trace return function: return function 1`] = `
"
function A(a, b, c) {
    if ( a === b) {
        return 'ok';
    }

    switch (c) {
        case '1':
            return '1';
    
        default:
            return '2';
    }

    return 'error';
}

      ↓ ↓ ↓ ↓ ↓ ↓

function A(a, b, c) {
  let __traceParent__ = window.__traceParent__;
  let __traceOldParent__ = __traceParent__;

  if (window.__trace__) {
    if (!__traceParent__.next) {
      __traceParent__.next = [];
    }

    const current = {
      name: 'A',
      params: [\\"a\\", \\"b\\", \\"c\\"]
    };

    __traceParent__.next.push(current);

    window.__traceParent__ = current;
  }

  if (a === b) {
    window.__traceParent__ = __traceOldParent__;
    return 'ok';
  }

  switch (c) {
    case '1':
      window.__traceParent__ = __traceOldParent__;
      return '1';

    default:
      window.__traceParent__ = __traceOldParent__;
      return '2';
  }

  window.__traceParent__ = __traceOldParent__;
  return 'error';
  window.__traceParent__ = __traceOldParent__;
}
"
`;
